pilot/pkg/model/context.go

```diff
@@ -88,6 +97,18 @@ type Proxy struct {
 
 	// Metadata key-value pairs extending the Node identifier
 	Metadata map[string]string
+
+	// mutex control access to mutable fields in the Proxy. On-demand will modify the
+	// list of services based on calls from envoy.
+	mutex sync.RWMutex
+
+	// serviceDependencies, if set, controls the list of outbound listeners and routes
+	// for which the proxy will receive configurations. If nil, the proxy will get config
+	// for all visible services.
+	// The list will be populated either from explicit declarations or using 'on-demand'
+	// feature, before generation takes place. Each node may have a different list, based on
+	// the requests handled by envoy.
+	serviceDependencies []*Service
```

Comment:
A question: how this field can be updated? I can not find.

Reply:
OnConnect